File: Nivedita/Linkedlist/Rotate a Linked List .cpp
class Solution
{
    public:
    //Function to rotate a linked list.
    Node* rotate(Node* head, int k)
    {
      Node*temp=head;
      int n=k;
      while(temp->next!=NULL)
      {
          temp=temp->next;
      }
      temp->next=head;
      Node*ptr;
      while(k--)
      {
          ptr=head;
          head=head->next;
      }
      ptr->next=NULL;
      return head;// Your code here
    }
};

File: JyotiKumari/Linked_list_GFG/Rotate a linked list.py
# Your task is to complete this function

'''

class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

'''


class Solution:
    
    #Function to rotate a linked list.
    def rotate(self, head, k):
        # code here
        if k == 0:
            return
        
        current = head
        count = 1
        
        while(count <k and current != None):
            current = current.next
            count += 1
    
        if current is None:
            return

        kNode = current

        while(current.next is not None):
            current = current.next

        current.next = head
        head = kNode.next
        kNode.next = None
        return head